### PR TITLE
[Fix] fix bugs in testing process of topdown heatmap models

### PR DESCRIPTION
**Pull request by Ben-Louis**
## Modification

1. manually set HSigmoid params
2. change the shape of `center`, `scale` to (1, 2) for (2,) in face datasets
3. manually convert `mask` to boolean array in `_calc_distances`
5. use `bbox_centers` and `bbox_scales` when `bboxes` is absent from `gt_instances` in `BaseHead.decode`

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/datasets/datasets/face/aflw_dataset.py [hunk 1]
```diff
@@ -81,8 +81,8 @@ def parse_data_info(self, raw_data_info: dict) -> Optional[dict]:
 
         img_path = osp.join(self.data_prefix['img'], img['file_name'])
 
-        center = np.array(ann['center'], dtype=np.float32)
-        scale = np.array([ann['scale'], ann['scale']], dtype=np.float32)
+        center = np.array(ann['center'], dtype=np.float32).reshape(1, 2)
```
Comment (ly015) [inline mmpose/datasets/datasets/face/aflw_dataset.py:84]: I suggest sticking to one style and mention the shape in comments:

```python
center = np.array([ann['center']], dtype=np.float32)  # shape: (1, 2)
scale = np.array([ann['scale'], ann['scale']], dtype=np.float32)  # shape: (1, 2)
```

File: mmpose/models/heads/base_head.py [hunk 2]
```diff
@@ -126,7 +126,13 @@ def decode(self, batch_outputs: Union[Tensor, Tuple[Tensor]],
             # Store the keypoint predictions in the data sample
             if 'pred_instances' not in data_sample:
                 pred_instances = InstanceData()
-                pred_instances.bboxes = data_sample.gt_instances.bboxes
+                if 'bboxes' in data_sample.gt_instances:
+                    pred_instances.bboxes = data_sample.gt_instances.bboxes
```
Comment (ly015) [inline mmpose/models/heads/base_head.py:130]: When will the `bboxes` be absent, and can `bboxes` be replaced by `bbox_centers` and `bbox_scales` in the metrics, visualizer or other places where `bboxes` is used?

File: mmpose/models/heads/base_head.py [hunk 3]
```diff
@@ -126,7 +126,13 @@ def decode(self, batch_outputs: Union[Tensor, Tuple[Tensor]],
...
+                else:
+                    pred_instances.bbox_center = \
+                        data_sample.gt_instances.bbox_centers
+                    pred_instances.bbox_scale = \
+                        data_sample.gt_instances.bbox_scales
```
Comment (ly015) [inline mmpose/models/heads/base_head.py:135]:
```suggestion
                    pred_instances.bbox_centers = \
                        data_sample.gt_instances.bbox_centers
                    pred_instances.bbox_scales = \
                        data_sample.gt_instances.bbox_scales
```

File: mmpose/models/heads/heatmap_heads/heatmap_head.py [hunk 4]
```diff
@@ -368,7 +368,7 @@ def _load_state_dict_pre_hook(self, state_dict, prefix, local_meta, *args,
             # have keys like "final_layer.n.xxx", where the weights of the last
             # one should be renamed "final_layer.xxx", and others should be
             # renamed "conv_layers.n.xxx"
-            k_parts = _k.split('.')
+            k_parts = k.split('.')
```
Comment (ly015) [inline mmpose/models/heads/heatmap_heads/heatmap_head.py:371]: Which models are affected by this bug?

File: mmpose/models/heads/base_head.py (hunk 2 above)
Comment (Ben-Louis) [reply mmpose/models/heads/base_head.py:130]: Some face datasets, such as 300w, wflw, and aflw, do not provide `bbox`. In these cases the `bboxes` will be absent in `gt_instances`. 
Now we only compute NME on these datasets, where `bbox_centers` and `bbox_scales` are not used. They might be required if users want to plot predicted keypoints on original images. Do we need to support this function in visualizer?

Comment (ly015) [reply mmpose/models/heads/base_head.py:130]: I suggest we add `bbox` in the data_info of these datasets, using `bbox_cs2xyxy`.

File: mmpose/models/heads/heatmap_heads/heatmap_head.py (hunk 4 above)
Comment (ly015) [reply mmpose/models/heads/heatmap_heads/heatmap_head.py:371]: Just wondering how HRNet survises this bug. And is HRNetv2's performance aligned with the record after fixing this or it just can load checkpoints without raising errors?

**Review by ly015 — APPROVED** [review]
